zqd/space/config.go

```diff
@@ -3,16 +3,27 @@ package space
 import (
 	"fmt"
 	"path/filepath"
-	"regexp"
 	"strings"
+	"unicode"
 
 	"github.com/brimsec/zq/pkg/fs"
 	"github.com/brimsec/zq/zqd/api"
 	"github.com/brimsec/zq/zqd/storage"
 	"github.com/brimsec/zq/zqe"
 )
 
-var nameRegexp = regexp.MustCompile("[^-.a-zA-Z0-9_]")
+func validSpaceNameRune(r rune) bool {
+	return r != '/' && unicode.IsPrint(r)
+}
+
+func validSpaceName(s string) bool {
+	for _, r := range s {
+		if !validSpaceNameRune(r) {
+			return false
+		}
+	}
+	return true
```

Comment:
Maybe change validSpaceNameRune to invalidSpaceNameRune and then simplify to this?
```go
return strings.IndexFunc(s, invalidSpaceNameRune) == -1
```